driver/src/main/com/mongodb/DBCollection.java

```diff
@@ -1067,17 +1066,8 @@ public List distinct(final String fieldName, final DBObject query) {
      */
     @SuppressWarnings("unchecked")
     public List distinct(final String fieldName, final DBObject query, final ReadPreference readPreference) {
-        BsonArray distinctArray = executor.execute(new DistinctOperation(getNamespace(), fieldName).filter(wrapAllowNull(query)),
-                                                   readPreference);
-
-        List distinctList = new ArrayList();
-        for (BsonValue value : distinctArray) {
-            BsonDocument document = new BsonDocument("value", value);
-            DBObject obj = getDefaultDBObjectCodec().decode(new BsonDocumentReader(document), DecoderContext.builder().build());
-            distinctList.add(obj.get("value"));
-        }
-
-        return distinctList;
+        return new OperationIterable<Object>(new DistinctOperation(getNamespace(), fieldName).filter(wrap(query)),
```

Comment:
This is the problem with hard-coding DocumentCodec in DistinctOperation, as this method can now return Document instances instead of DBObject instances.